src/main/java/uk/ac/ebi/eva/commons/models/converters/data/DBObjectToVariantConverter.java

```diff
@@ -0,0 +1,250 @@
+//                .append(IDS_FIELD, object.getIds())    //Do not include IDs.
+                .append(TYPE_FIELD, object.getType().name())
+                .append(CHROMOSOME_FIELD, object.getChromosome())
+                .append(START_FIELD, object.getStart())
+                .append(END_FIELD, object.getEnd())
+                .append(LENGTH_FIELD, object.getLength())
+                .append(REFERENCE_FIELD, object.getReference())
+                .append(ALTERNATE_FIELD, object.getAlternate());
+
+        // Internal fields used for query optimization (dictionary named "_at")
+        BasicDBObject _at = new BasicDBObject();
+        mongoVariant.append("_at", _at);
+        
+        // ChunkID (1k and 10k)
+        String chunkSmall = object.getChromosome() + "_" + object.getStart() / VariantMongoDBWriter.CHUNK_SIZE_SMALL + "_" + VariantMongoDBWriter.CHUNK_SIZE_SMALL / 1000 + "k";
+        String chunkBig = object.getChromosome() + "_" + object.getStart() / VariantMongoDBWriter.CHUNK_SIZE_BIG + "_" + VariantMongoDBWriter.CHUNK_SIZE_BIG / 1000 + "k";
+        BasicDBList chunkIds = new BasicDBList(); chunkIds.add(chunkSmall); chunkIds.add(chunkBig);
+        _at.append("chunkIds", chunkIds);
+        
+        // Transform HGVS: Map of lists -> List of map entries
+        BasicDBList hgvs = new BasicDBList();
+        for (Map.Entry<String, Set<String>> entry : object.getHgvs().entrySet()) {
+            for (String value : entry.getValue()) {
+                hgvs.add(new BasicDBObject(TYPE_FIELD, entry.getKey()).append(NAME_FIELD, value));
+            }
+        }
+        mongoVariant.append(HGVS_FIELD, hgvs);
+        
+        // Files
+        if (variantSourceEntryConverter != null) {
+            BasicDBList mongoFiles = new BasicDBList();
+            for (VariantSourceEntry archiveFile : object.getSourceEntries().values()) {
+                mongoFiles.add(variantSourceEntryConverter.convertToStorageType(archiveFile));
+            }
+            mongoVariant.append(FILES_FIELD, mongoFiles);
+        }
+        
+//        // Annotations
+//        if (variantAnnotationConverter != null) {
+//            if (object.getAnnotation() != null) {
+//                DBObject annotation = variantAnnotationConverter.convertToStorageType(object.getAnnotation());
+//                mongoVariant.append(ANNOTATION_FIELD, annotation);
+//            }
+//        }
+
+        // Statistics
+        if (statsConverter != null) {
+            List mongoStats = statsConverter.convertCohortsToStorageType(object.getSourceEntries());
+            mongoVariant.put(STATS_FIELD, mongoStats);
+        }
+        return mongoVariant;
+    }
+
+    public String buildStorageId(Variant v) {
+        return buildStorageId(v.getChromosome(), v.getStart(), v.getReference(), v.getAlternate());
+    }
+
+    public String buildStorageId(String chromosome, int start, String reference, String alternate) {
```

Comment:
fixed in next commit. We may need to move that method to variation commons along with the converters